parking_lot.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from typing import Optional, Union
+from enum import Enum
+from pydantic import BaseModel, Field, ValidationError, validator
+from fastapi import FastAPI, Request, status
+from starlette.responses import JSONResponse
+import math
+import dateutil.parser
+
+app = FastAPI()
+
```

Comment:
What ideas do you have to monitor the service?

Let's say your application is running  in a production environment,  an a failure occurs, how can you determine the impact of the failure and find out the reason of failure?

What fault tolerance mechanism could be implemented to avoid this kind of problems in advance and provide a 100% use availability ?

Reply:
New Relic maybe ?